gneiss/regression/_ols.py

```diff
@@ -192,10 +185,110 @@ def fit(self, **kwargs):
         ----------
         **kwargs : dict
            Keyword arguments used to tune the parameter estimation.
+        """
```

Comment:
I think it will be worth having some documentation of what we are doing on each step so we can follow the description and code.